handle corner cases when reading from stdin

midichef opened the pull request:
The first commit makes it possible to use stdin as a replay file:  `vd -p -`.
Right now that causes a traceback:
```
Traceback (most recent call last):
...
File "/home/midichef/.local/lib/python3.10/site-packages/visidata/main.py", line 347, in main_vd
    vdfile.name = 'stdin.vd'
AttributeError: can't set attribute 'name'
```

The second commit is for interactive use of visidata. When a user runs `open-file` and inputs `-` as the filename, visidata hangs. The patch catches that and gives an error.

midichef committed:
[open-] fix hang after open-file runs on tty stdin

midichef committed:
[main- path-] make "-p -" replay stdin as a .vdj file

saulpw reviewed (approved):
Thanks for this, @midichef !

midichef commented:
I fixed one more case, the case of stdin for replay commands, `-p -`, but with another file argument present, `a.json`:
```
cat quit.vdj |vd -p - a.json
    Traceback (most recent call last):
    ...
    File "/home/midichef/.local/lib/python3.10/site-packages/visidata/main.py", line 46, in eval_vd
        log = log.format(*args, **kwargs)
    KeyError: '"sheet"'
```
by adding this line:
https://github.com/saulpw/visidata/blob/32a59948f63b75011cb1f55bfd07ba9268924326/visidata/main.py#L40
And a previous commit message had a typo, so I fixed it and force-pushed a replacement.

anjakefala commented:
When testing, I found an additional cornercase bug!

```
cat tests/append.vd | vd -p -  
```

Currently, VisiData expects `.vdj` from stdin

```
message	source
opening - as vdj	visidata/_open.py:138:openPath
JSONDecodeError: Expecting value: line 1 column 1 (char 0)	visidata/errors.py:26:exceptionCaught

```

anjakefala commented:
@midichef I am going to address that new cornercase in this PR!

anjakefala committed:
[docs] document replay support from stdin for .vdj

anjakefala commented:
Hi @midichef! 

It ended up being 15 lines of code to support both `.vd` and `.vdj` from stdin, so we're just going to merge as-is!
